File: 523.continuous-subarray-sum.java
public class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        if (nums == null || nums.length == 0)
            return false;

        int[] preSum = new int[nums.length + 1];

        for (int i = 1; i <= nums.length; i++) {
            preSum[i] = preSum[i - 1] + nums[i - 1];
        }

        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 2; j <= nums.length; j++) {
                if (k == 0) {
                    if (preSum[j] - preSum[i] == 0) {
                        return true;
                    }
                } else if ((preSum[j] - preSum[i]) % k == 0) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: 523.continuous-subarray-sum.js
/*
 * @lc app=leetcode id=523 lang=javascript
 *
 * [523] Continuous Subarray Sum
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var checkSubarraySum = function (nums, k) {
  map = {
    0: -1,
  };

  let runningSum = 0;

  // Loop这个nums
  for (let i = 0; i < nums.length; i++) {
    runningSum += nums[i];
    console.log(runningSum);
    if (k != 0) runningSum %= k;
    console.log("After mod k:" + runningSum);
    let prev = map[runningSum];
    if (prev != null) {
      console.log("Prev is not null and let's see the prev:" + prev);
      if (i - prev > 1) continue;
    } else map[runningSum] = i;
    console.log(map);
  }
  return false;
};

checkSubarraySum([23, 2, 4, 6, 7], 6);
// @lc code=end
